src/snyk/snykOss/services/ossService.ts

```diff
@@ -45,18 +48,35 @@ export class OssService extends CliService<OssResult> {
     return result;
   }
 
-  protected beforeTest(): void {
+  protected beforeTest(manualTrigger: boolean, reportTriggeredEvent: boolean): void {
```

Comment:
This is not the config check (as in `itly.ts`), rather check if event was triggered in previous code logic.

The reason for that is because there's a possibility to run all scans (OSS, Code) together or specifically via watchers (only OSS or only Code). When all scans are run together we have to prepare an array of triggered analysis types (see `snykLib.ts`). This means we shouldn't send an already sent event in this place (when single analysis type is triggered).

I don't like that event sending logic is scattered in several places now 😢 but that's the cleanest solution I could come up with. Any better idea?

Reply:
> run all scans (OSS, Code) together or specifically via watchers (only OSS or only Code)

☝️ that peace was missing for me.

I think it's ok for now. No better idea comes in our mind.